File: src/Blogorian.Data/Route.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Blogorian.Data
{
    public class Route
    {
        public string Path { get; set; }
        public string TemplateUrl { get; set; }
        public string Controller { get; set; }
        public string ControllerAs { get; set; }

        public Route(string path, string templateUrl, string controller, string controllerAs)
        {
            this.Path = path;
            this.TemplateUrl = templateUrl;
            this.Controller = controller;
            this.ControllerAs = controllerAs;
        }
    }
}

File: src/Blogorian.Web/wwwroot/app/core/route.ts
module Blogorian.Core.Route {
    import ngRoute = angular.route;
    export class BlogorianRoute implements angular.route.IRoute {
        path: string;
        templateUrl: string;
        controller: string | Function;
        controllerAs: string;
        constructor(path: string, templateUrl: string, controller: string | Function, controllerAs?: string) {
            this.path = path;
            this.templateUrl = templateUrl;
            this.controller = controller;
            if (controllerAs)
                this.controllerAs = controllerAs;
        }
    }
}